spread/client.go

```diff
@@ -79,14 +79,29 @@ func (c *Client) dialOnReboot() error {
 
 	waitConfig := *c.config
 	waitConfig.Timeout = 5 * time.Second
+	uptimeDelta := 3 * time.Second
+
 	for {
 		before := time.Now()
 		sshc, err := ssh.Dial("tcp", c.addr, &waitConfig)
 		if err != nil {
 			// It's gone.
-			break
+			continue
+		}
+
+		c.sshc.Close()
+		c.sshc = sshc
+		currUptime, err := c.getUptime()
+		if err != nil {
+			// Not available uptime yet
+			continue
+		}
+
+		elapsedTime := currUptime.Sub(prevUptime)
```

Comment:
Elapsed makes it sound like the time since we started looking, but this is really the elapsed time between one boot and the other boot, so this is actually the real _uptimeDelta._ The test then becomes more readable:
```
if uptimeDelta > uptimeChanged { ... }
```